event-based-simulators/extras/ArgumentsAndCredentialsHandler.py

```diff
@@ -0,0 +1,186 @@
+
+    baseurl = args.baseurl
+    if not baseurl:
+        baseurl = Environment.C8Y_BASE
+    baseurl = RemoveTrailingSlashFromBaseUrl(baseurl)
+
+    tenant = args.tenant_id
+    if not tenant:
+        tenant = Environment.C8Y_TENANT
+
+    c8y = CumulocityApi(base_url=baseurl,  # the url of your Cumulocity tenant here
+                        tenant_id=tenant,  # the tenant ID of your Cumulocity tenant here
+                        username=username,  # your Cumulocity IoT username
+                        password=password)  # your Cumulocity IoT password
+
+    return data_type, device_ids_list, create_from, create_to, log_level, c8y
+
+
+def HandleImportArguments():
+    parser = argparse.ArgumentParser(description='Script to import profiles data')
+    parser.add_argument('--ifiles', '-i', type=str, help='Input file', nargs='+')
+    parser.add_argument('--log', '-l', choices=['DEBUG', 'INFO', 'WARNING', 'ERROR', 'CRITICAL'], help='Log-level')
+    parser.add_argument('--username', '-u', type=str, help='C8Y Username')
+    parser.add_argument('--password', '-p', type=str, help='C8Y Password')
+    parser.add_argument('--baseurl', '-b', type=str, help='C8Y Baseurl')
+    parser.add_argument('--tenant-id', '-t', type=str, help='C8Y TenantID')
+
+    args = parser.parse_args()
+
+    if not os.path.exists("export_data"):
+        logging.info("No export_data folder found")
+        sys.exit()
+    INPUT_FILE_LIST = args.ifiles
+    if not INPUT_FILE_LIST:
+        INPUT_FILE_LIST = []
+
+    LOG_ARGUMENT = args.log
+    LOG_LEVEL = logging.INFO
+    if LOG_ARGUMENT == 'DEBUG':
+        LOG_LEVEL = logging.DEBUG
+    elif LOG_ARGUMENT == 'WARNING':
+        LOG_LEVEL = logging.WARNING
+    elif LOG_ARGUMENT == 'ERROR':
+        LOG_LEVEL = logging.ERROR
+    elif LOG_ARGUMENT == 'CRITICAL':
+        LOG_LEVEL = logging.CRITICAL
+
+    USERNAME = args.username
+    if not USERNAME:
+        USERNAME = Environment.C8Y_USER
+
+    PASSWORD = args.password
+    if not PASSWORD:
+        PASSWORD = Environment.C8Y_PASSWORD
+
+    BASEURL = args.baseurl
+    if not BASEURL:
+        BASEURL = Environment.C8Y_BASE
+    BASEURL = RemoveTrailingSlashFromBaseUrl(BASEURL)
+
+    TENANT = args.tenant_id
+    if not TENANT:
+        TENANT = Environment.C8Y_TENANT
+
+    c8y = CumulocityApi(base_url=BASEURL,  # the url of your Cumulocity tenant here
+                        tenant_id=TENANT,  # the tenant ID of your Cumulocity tenant here
+                        username=USERNAME,  # your Cumulocity IoT username
+                        password=PASSWORD)  # your Cumulocity IoT password
+
+    return INPUT_FILE_LIST, LOG_LEVEL, c8y
+
+
+def RemoveTrailingSlashFromBaseUrl(baseUrl):
+    if baseUrl[-1] == '/':
+        return baseUrl[:-1]
+    return baseUrl
+
+
+def CheckTenantConnection(baseUrl, C8Y_HEADERS):
+    # Check if connection to tenant can be created
+    try:
+        requests.get(f'{baseUrl}/tenant/currentTenant', headers=C8Y_HEADERS)
```

Comment:
I would suggest to also check the response code: in case of 40x (or "not ok") , e.g. with invalid credentials, the response should be printed and the connection should be handled as failed!